### PR TITLE
Add debug collision shape to CSG with collision

### LINKED ISSUE #58952: Visible Collision Shapes doesn't draw CSG collisions
### Godot version

3.4, 4.0

### System information

macOS, Windows

### Issue description

Visible Collision Shapes isn't drawing CSG collisions making it harder to debug physics problems

### Steps to reproduce

Enable Debug > Visible Collision Shapes

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by rburing**
Before:

csg_without_collision

After:

csg_with_collision

This also makes it clearer that CSG's use trimesh collision, which doesn't seem to be obvious to everyone.

- Fixes https://github.com/godotengine/godot/issues/58952

### CODE REVIEW

File: modules/csg/csg_shape.cpp [hunk 1]
```diff
@@ -567,13 +604,17 @@ void CSGShape3D::_notification(int p_what) {
 				PhysicsServer3D::get_singleton()->free(root_collision_instance);
 				root_collision_instance = RID();
 				root_collision_shape.unref();
+				if (_is_debug_collision_shape_visible()) {
+					_clear_debug_collision_shape();
+				}
```
Comment (kleonc) [inline modules/csg/csg_shape.cpp:609]: I think toggling debug collision off (by calling `SceneTree::set_debug_collisions_hint(false)`) can lead to a leak, `root_collision_debug_instance` could end up not being cleared in such case.
```suggestion
				_clear_debug_collision_shape();
```

**Comment by rburing** [comment]
@kleonc Thanks, done.

I also made the debug shape show up in the editor (as is generally expected for collision shapes). I think it will improve users' understanding of how CSG collision [works](https://github.com/godotengine/godot/issues?q=is%3Aissue+is%3Aopen+csg+label%3Atopic%3Aphysics).

**Review by clayjohn — APPROVED** [review]
The RenderingServer API usage looks fine to me!

**Review by akien-mga — APPROVED** [review]
Looks good to me. I've been missing this feature when playing with CSG myself :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**